pkg/bucket/lifecycle/lifecycle.go

```diff
@@ -160,15 +169,26 @@ func (lc Lifecycle) FilterActionableRules(obj ObjectOpts) []Rule {
 			continue
 		}
 		// The NoncurrentVersionExpiration action requests MinIO to expire
-		// noncurrent versions of objects 100 days after the objects become
+		// noncurrent versions of objects x days after the objects become
 		// noncurrent.
 		if !rule.NoncurrentVersionExpiration.IsDaysNull() {
 			rules = append(rules, rule)
 			continue
 		}
+		// The NoncurrentVersionTransition action requests MinIO to transition
+		// noncurrent versions of objects x days after the objects become
+		// noncurrent.
+		if !rule.NoncurrentVersionTransition.IsDaysNull() {
```

Comment:
@poornas the point of FilterActionableRules is return rules that match prefix & tags, rules that could be actionable. I know that function name is not good, but it is just easier move the decision outside this function

Reply:
transition rules can also come with prefix and tags - so made sense to do this check within this function